### DISCUSSION TITLE
Deployment on Vercel fails from time to time

### DISCUSSION THREAD

**Discussion by BrandonStudio** [lobehub/lobehub#1552] [Self Hosting]
`bun install` fails from time to time.
![image](https://github.com/lobehub/lobe-chat/assets/55647556/d9fdc009-6607-4b08-b35f-82538e90e765)

<details>
<summary>Deployments</summary>

![image](https://github.com/lobehub/lobe-chat/assets/55647556/b4f76865-b2d4-459a-9fa2-f44e261f866c)

</details>

What is the problem?

**Comment by arvinxx** [comment]
Just try again

**Reply by BrandonStudio** [reply]
Indeed.
But I would like to know why this happens, and how to prevent it.
Do you think this is a problem of bun or Vercel?

**Reply by arvinxx** [reply]
I think it might be a problem with bun. pnpm would be more stable in my mind

**Reply by arvinxx** [reply]
> how to prevent it.

maybe you can set the install scripts to `pnpm i` in `vercel.json` 

https://github.com/lobehub/lobe-chat/blob/260caa3cb7722a61bf56f4e29b47254a8f1398b6/vercel.json#L2

I perfer to stay with bun officially as installing dependencies with bun is really fast.

**Reply by BrandonStudio** [reply]
@arvinxx Do you consider delete `vercel.json`? As this kind of property can be set through Vercel dashboard, and not all users use Vercel to host, which makes this file unnecessary.

**Reply by arvinxx** [reply]
I will keep `vercel.json`, because not all users know to how to config the vercel dashboard. I can't insure our self-hosting users can get a fast and correct deployment with vercel's the default config.

Beside, with `vercel.json`, I can keep control to avoid deployment failed like this: https://github.com/lobehub/lobe-chat/pull/960

**Comment by cy948** [comment]
You can overwrite it directly in vercel dashboard without changing the `vercel.json`.
![image](https://github.com/lobehub/lobe-chat/assets/67412196/7fe430dc-f77d-4b34-a331-84cfaf8cd3d1)

**Reply by BrandonStudio** [reply]
This does not work in practice for production branch.

**Reply by cy948** [reply]
Fine, consider fork the repository than delete it in your repo? Enable the auto-sync action to get updates.
